portfolio/src/main/webapp/script.js

```diff
@@ -32,17 +32,22 @@ function addRandomFact() {
  * Fetch the list of comments and add them to the page
  */
 function fetchAndAddComments(){
-  fetch('/comments').then(response => response.json()).then((comments) => {
+  fetch('/comments')
+  .then(response => response.json())
+  .then((comments) =>{
     const commentContainer = document.getElementById('comments-container');
-    comments.forEach(function(comment) {
-      commentContainer.appendChild(createListElement(comment))
-    });
+    for (i in comments) {
+        commentContainer.appendChild(createCommentElement(comments[i]));
+    }
```

Comment:
you could have kept the forEach. 
comments.forEach(function(comment) {
  // Do something with coment
});

Does this not work?

Reply:
Depends on how the JSON is returned I think. IIRC this ends up being an index.